### DISCUSSION TITLE
How do I add text inside a path drawing in Fabric.js?

### DISCUSSION THREAD

**Discussion by ubhoopathy** [fabricjs/fabric.js#9667] [Q&A]
If use this code
let path = new fabric.Path('M 0 0 A 80 80 0 0 1 170 0 M 0 0 A 20 20 0 0 0 170 0', {
    fill: '',
    strokeWidth: 1,
    stroke: '',
    visible: true,
  });
  const text = new fabric.Text('New Curve Text New Curve Text', {
    ...defaultOptions,
    fontSize: 40,
    editable: false,
    path: path,
  });
![Screenshot 2024-02-14 105100](https://github.com/fabricjs/fabric.js/assets/149072388/4f18c232-26e2-4927-a81f-c0d8c8b24255)

I'm expecting results. below
![pathtext2](https://github.com/fabricjs/fabric.js/assets/149072388/99cd27d6-c285-42b4-a337-efde9e881351)
![pathtext1](https://github.com/fabricjs/fabric.js/assets/149072388/7fee8a85-6957-4968-a4f7-40161bfc4166)

**Comment by ShaMan123** [comment]
Fabric doesn't support this (for now at least)
You have 2 options that are almost identical in requirements:

1. Calculate a path that describes the text layout you show, with line breaks and all. I would use intersection logic to do that:
    - According to font size intersect your shape
    - For each line add to your path a `moveTo` `lineTo`
    - Take into account the shape's edge angles
3. Override text layout methods of Textbox, start here: https://github.com/fabricjs/fabric.js/blob/2529ecf63b3af5a879ece846ebf572a379c51edf/src/shapes/Textbox.ts#L306

I tend to think the first option might be easier but I am not sure
Anyways, poly intersection can be done by fabric's `Intersection` class
Oval intersection is pretty simple to do
Path intersection require `bezier-js`

Sounds cool
In case you are going open source with this I might be able to provide the intersection logic. That is the big part of the work.

**Reply by ShaMan123** [reply]
You might need to use both options to impl all functionality such as text alignment

**Reply by ShaMan123** [reply]
3rd options which is a mix of both.
In the Textbox layout methods when a char is measured check if it is in the shape and adjust it accordingly
